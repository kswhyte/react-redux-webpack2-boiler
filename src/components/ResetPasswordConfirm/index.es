import Component from './ResetPasswordConfirm';
export default Component;
