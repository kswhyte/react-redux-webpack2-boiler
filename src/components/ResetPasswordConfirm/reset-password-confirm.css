@import "imports/_helpers.css";
@import "imports/_layout.css";
@import "imports/_form-elements.css";
