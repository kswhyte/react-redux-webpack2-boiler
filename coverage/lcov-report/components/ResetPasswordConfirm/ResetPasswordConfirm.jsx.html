<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/ResetPasswordConfirm/ResetPasswordConfirm.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/ResetPasswordConfirm</a> ResetPasswordConfirm.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>3/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>3/3</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
// import { Link } from 'react-router-dom';
&nbsp;
import HgButton from '@hg/three-ui/HgButton';
&nbsp;
import './reset-password-confirm.css';
&nbsp;
const ResetPasswordConfirm = () =&gt; {
  return (
    &lt;div className="reset-password-confirm-wrapper"&gt;
      &lt;h2&gt;RESET YOUR PASSWORD&lt;/h2&gt;
      &lt;h3&gt;
        An email was sent to you with further instructions on&lt;br /&gt;
        how to retrieve your password.
      &lt;/h3&gt;
&nbsp;
      &lt;div className="check-confirm-icon" /&gt;
&nbsp;
      &lt;form&gt;
        &lt;HgButton text="SIGN IN" /&gt;
      &lt;/form&gt;
    &lt;/div&gt;
  );
};
&nbsp;
ResetPasswordConfirm.propTypes = {};
&nbsp;
export default ResetPasswordConfirm;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Aug 09 2017 12:57:20 GMT-0600 (MDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
