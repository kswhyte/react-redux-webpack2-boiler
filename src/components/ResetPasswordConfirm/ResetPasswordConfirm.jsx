import React from 'react';
import { Link } from 'react-router-dom';

import './reset-password-confirm.css';

const ResetPasswordConfirm = () => {
  return (
    <div className="reset-password-confirm-wrapper card">
      <img src="images/pattern-dots.png" className="dots-pattern" />
      <h2>RESET YOUR PASSWORD</h2>
      <div className="page-instructions col-xs-offset-3 col-xs-6">
        <p>
          An email was sent to you with further instructions on<br />
          how to retrieve your password.
        </p>
      </div>

      <div className="row lg-spacer">
        <div className="col-xs-offset-3 col-xs-6">
          <span className="lnr lnr-checkmark-circle" />
        </div>
      </div>

      <form>
        <div className="row sm-spacer">
          <div className="col-xs-offset-3 col-xs-6">
            <Link to="/login" id="back-to-signin-link" className="primary-btn btn-link">
              SIGN IN
            </Link>
          </div>
        </div>
      </form>
    </div>
  );
};

ResetPasswordConfirm.propTypes = {};

export default ResetPasswordConfirm;
