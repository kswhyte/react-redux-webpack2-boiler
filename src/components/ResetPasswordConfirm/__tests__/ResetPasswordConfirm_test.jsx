import React from 'react';
// import { storiesOf, action, linkTo } from '@kadira/storybook';
import ResetPasswordConfirm from '../ResetPasswordConfirm';

// import { specs, describe, it } from 'storybook-addon-specifications';

import { storiesOf, describe, it, specs } from '../../../../.storybook/facade';

import { shallow, mount } from 'enzyme';
import { StaticRouter } from 'react-router';
import expect from 'expect';

const stories = storiesOf('ResetPasswordConfirm', module);
const props = {
}
stories.add('ResetPasswordConfirm Page Story', () => {
  const resetPasswordConfirmStory = (
    <StaticRouter {...props} context={{}}>
      <ResetPasswordConfirm {...props} />
    </StaticRouter>
  );
  specs(() =>
    describe('ResetPasswordConfirm page and functionality', () => {
      it('Should render the ResetPasswordConfirm component without crashing', () => {
        let output = mount(resetPasswordConfirmStory);
        expect(output.find('.reset-password-confirm-wrapper').exists()).toBe(true)
      });
      it('Should have a header element that says "Reset Your Password"', () => {
        let output = mount(resetPasswordConfirmStory);
        expect(output.find('h2').text()).toContain('RESET YOUR PASSWORD');
      });
      it('Should have a <p> tag that says "An email was sent to you with further instructions on how to retrieve your password."', () => {
        let output = mount(resetPasswordConfirmStory);
        expect(output.find('p').text()).toContain('An email was sent to you with further instructions onhow to retrieve your password.');
      });
      it('Should have one span icon/font from linearicons', () => {
        let output = mount(resetPasswordConfirmStory);
        expect(output.find('.lnr-checkmark-circle').length).toEqual(1);
      });
      it('Should have a "Sign On" button to return to login', () => {
        let output = mount(resetPasswordConfirmStory);
        expect(output.find('a').length).toEqual(1);
        expect(output.find('.primary-btn').length).toEqual(1);
      });
    })
  );
  return resetPasswordConfirmStory;
});
