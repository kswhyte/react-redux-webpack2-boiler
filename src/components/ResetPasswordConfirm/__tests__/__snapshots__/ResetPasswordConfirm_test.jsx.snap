// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResetPasswordConfirm Page Story 1`] = `
<div
  className="reset-password-confirm-wrapper card"
>
  <img
    className="dots-pattern"
    src="images/pattern-dots.png"
  />
  <h2>
    RESET YOUR PASSWORD
  </h2>
  <div
    className="page-instructions col-xs-offset-3 col-xs-6"
  >
    <p>
      An email was sent to you with further instructions on
      <br />
      how to retrieve your password.
    </p>
  </div>
  <div
    className="row lg-spacer"
  >
    <div
      className="col-xs-offset-3 col-xs-6"
    >
      <span
        className="lnr lnr-checkmark-circle"
      />
    </div>
  </div>
  <form>
    <div
      className="row sm-spacer"
    >
      <div
        className="col-xs-offset-3 col-xs-6"
      >
        <a
          className="primary-btn btn-link"
          href="/login"
          id="back-to-signin-link"
          onClick={[Function]}
        >
          SIGN IN
        </a>
      </div>
    </div>
  </form>
</div>
`;
